src/XMakeBuildEngine/UnitTestsPublicOM/Definition/Project_Tests.cs

```diff
@@ -2415,6 +2418,425 @@ public void ImportPropertyEvaluatingToInvalidPath()
+
+        [Fact]
+        public void GetItemProvenanceWhenExcludeHasIndirectReferences()
+        {
+            var project =
+                @"<Project ToolsVersion='msbuilddefaulttoolsversion' DefaultTargets='Build' xmlns='msbuildnamespace'>
+                  <ItemGroup>
+                    <B Include=`1;2;3`/>
+                    <A Include=`1;2;3` Exclude=`$(P);@(B)`/>
+                  </ItemGroup>
+
+                  <PropertyGroup>
+                    <P>1;2;3;@(B)</P>  
+                  </PropertyGroup>
+                </Project>
+                ";
+
+            var expected = new ProvenanceResultTupleList
+            {
+                Tuple.Create("B", Operation.Include, Provenance.StringLiteral, 1),
+                Tuple.Create("A", Operation.Exclude, Provenance.Inconclusive | Provenance.StringLiteral, 3)
+            };
+
+            AssertProvenanceResult(expected, project, "1");
+        }
+
+        [Fact]
+        public void GetItemProvenanceWhenIncludeHasIndirectReferences()
+        {
+            var project =
+                @"<Project ToolsVersion='msbuilddefaulttoolsversion' DefaultTargets='Build' xmlns='msbuildnamespace'>
+                  <ItemGroup>
+                    <B Include=`1;2;3`/>
+                    <A Include=`$(P);@(B)`/>
+                    <C Include=`@(A)`/>
+                  </ItemGroup>
+
+                  <PropertyGroup>
+                    <P>1;2;3;@(B)</P>  
+                  </PropertyGroup>
+                </Project>
+                ";
+
+            var expected = new ProvenanceResultTupleList
+            {
+                Tuple.Create("B", Operation.Include, Provenance.StringLiteral, 1),
+                Tuple.Create("A", Operation.Include, Provenance.Inconclusive | Provenance.StringLiteral, 3),
+                Tuple.Create("C", Operation.Include, Provenance.Inconclusive, 3)
+            };
+
+            AssertProvenanceResult(expected, project, "1");
+        }
+
+        [Fact]
+        public void GetItemProvenanceWhenIncludeHasIndirectItemReferencesAndOnlyGlobsExistDirectly()
+        {
+            var project =
+                @"<Project ToolsVersion='msbuilddefaulttoolsversion' DefaultTargets='Build' xmlns='msbuildnamespace'>
+                  <ItemGroup>
+                    <B Include=`1;2;3`/>
+                    <A Include=`*;$(P);@(B)`/>
+                  </ItemGroup>
+
+                  <PropertyGroup>
+                    <P>@(B)</P>  
+                  </PropertyGroup>
+                </Project>
+                ";
+
+            var expected = new ProvenanceResultTupleList
+            {
+                Tuple.Create("B", Operation.Include, Provenance.StringLiteral, 1),
+                Tuple.Create("A", Operation.Include, Provenance.Inconclusive | Provenance.Glob, 3)
+            };
+
+            AssertProvenanceResult(expected, project, "1");
+        }
+
+        [Fact]
+        public void GetItemProvenanceShouldRespectItemConditions()
```

Comment:
This is interesting--is CPS on board with this behavior? It's nice and consistent, but I think I can imagine reasons to want to know provenance for a file that's not in the current configuration.